src/arch-clap.ts

```diff
@@ -0,0 +1,518 @@
+        sourceMap: true,
+        declaration: true,
+        types: ["node"]
+      },
+      include: ["src"]
+    };
+  }
+
+  loadAppPkg() {
+    this.appPkg = readAppPkgJson();
+    this.existAppPkgData = JSON.stringify(this.appPkg);
+    this.hasEslint = this.appHasDevDeps(...Object.keys(eslintDevDeps));
+    this.hasTypeScript = this.appHasDevDeps(...Object.keys(typeScriptDevDeps));
+  }
+
+  setupXclapFile() {
+    if (this.hasTypeScript) {
+      Fs.writeFileSync(
+        Path.resolve("xclap.ts"),
+        `import { loadTasks } from "@xarc/module-dev";
+loadTasks();
+`
+      );
+    } else {
+      Fs.writeFileSync(
+        Path.resolve("xclap.js"),
+        `require("@xarc/module-dev")();
+`
+      );
+    }
+  }
+
+  addDevDeps(...features: string[]) {
+    const isTs = features.includes("typescript");
+    const isEslint = features.includes("eslint");
+
+    if (isTs) {
+      this.addDevDepsToAppPkg(typeScriptDevDeps);
+    }
+
+    if (isEslint) {
+      this.addDevDepsToAppPkg(eslintDevDeps);
+    }
+
+    if (isTs && isEslint) {
+      this.addDevDepsToAppPkg(eslintTSDevDeps);
+    }
+
+    console.log(`INFO: dependencies added to your package.json, please install modules again.
+  ie: run 'npm install' for npm`);
+
+    this.saveAppPkgJson();
+  }
+
+  rmDevDeps(...features: string[]) {
+    const isTs = features.includes("typescript");
+    const isEslint = features.includes("eslint");
+
+    if (isTs) {
+      this.rmDevDepsFromAppPkg(typeScriptDevDeps);
+    }
+
+    if (isEslint) {
+      this.rmDevDepsFromAppPkg(eslintDevDeps);
+    }
+
+    if (isTs || isEslint) {
+      this.rmDevDepsFromAppPkg(eslintTSDevDeps);
+    }
+
+    console.log(`INFO: dependencies removed from your package.json. please install modules again.
+  ie: run 'npm install' for npm`);
+
+    this.saveAppPkgJson();
+  }
+
+  saveAppPkgJson(): boolean {
+    const newData = JSON.stringify(this.appPkg);
+    if (this.existAppPkgData !== newData) {
+      this.existAppPkgData = newData;
+      writeAppPkgJson(this.appPkg);
+      return true;
+    }
+    return false;
+  }
+
+  addDevDepsToAppPkg(dev: any) {
+    const devDep = this.appPkg.devDependencies || {};
+    for (const k in dev) {
+      devDep[k] = dev[k];
+    }
+    this.appPkg.devDependencies = sortObjKeys(devDep);
+  }
+
+  rmDevDepsFromAppPkg(dev: any) {
+    const devDep = this.appPkg.devDependencies || {};
+    for (const k in dev) {
+      delete devDep[k];
+    }
+    if (Object.keys(devDep).length === 0) {
+      delete this.appPkg.devDependencies;
+    } else {
+      this.appPkg.devDependencies = sortObjKeys(devDep);
+    }
+  }
+
+  appHasDevDeps(...deps: string[]): boolean {
+    return deps.every(x => {
+      const dd = this.appPkg.devDependencies;
+      return dd && dd.hasOwnProperty(x);
+    });
+  }
+
+  lintTask(dir: string): string[] {
+    const scanned = filterScanDir.sync({
+      dir,
+      grouping: true,
+      filter(file, path, extras: any) {
```

Comment:
Generally you don't want to add type annotations to callbacks and instead allow TS to infer the type based off the type definition of the function you're calling.